src/figuresClasses.ts

```diff
@@ -1,19 +1,70 @@
+type Shape = 'triangle'| 'circle' | 'rectangle';
+type Color = 'red'| 'green' | 'blue';
+
 export interface Figure {
+  shape: Shape;
+  color: Color;
 
+  getArea(): number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: Shape = 'triangle';
 
-}
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0 || a + b <= c || a + c <= b || c + b <= a) {
+      throw new Error(`sides ${a}, ${b} and ${c} can not form a triangle`);
+    }
+  }
 
-export class Circle {
+  getArea(): number {
+    const { a, b, c } = this;
+    const half = (a + b + c) / 2;
 
+    return +Math.sqrt(half * (half - a) * (half - b) * (half - c)).toFixed(2);
+  }
 }
 
-export class Rectangle {
+export class Circle implements Figure {
+  shape: Shape = 'circle';
+
+  constructor(
+    public color: Color,
+    public radius: number,
+  ) {
+    if (radius <= 0) {
+      throw new Error(`radius ${radius}can not form a circle`);
+    }
+  }
 
+  getArea(): number {
+    return +(Math.floor(Math.PI * (this.radius ** 2) * 100) / 100).toFixed(2);
+  }
 }
```

Comment:
```suggestion
  getArea(): number {
    return +(Math.floor(Math.PI * (this.radius ** 2) * 100) / 100);
  }
}
```
I think you don't need it here